agent/lib_zend_http.c

```diff
@@ -436,7 +436,7 @@ NR_PHP_WRAPPER_END
 void nr_zend_http_enable(TSRMLS_D) {
   if ((NR_FW_ZEND != NRPRG(current_framework))
       && (NR_FW_LAMINAS3 != NRPRG(current_framework))) {
-    nr_php_wrap_user_function(NR_PSTR(http_client_request),
+    nr_php_wrap_user_function(NR_PSTR(HTTP_CLIENT_REQUEST_Z),
```

Comment:
Was the failure that led to this PR due to `http_client_request` being set to the wrong value here? If so, `http_client_request` is used elsewhere in this file and may also be the incorrect value there as well. Additionally, if `http_client_request` is the wrong value, could `http_client` be set to the wrong value as well?

If I'm reading this code correctly, `http_client_request` should never be set to the wrong value here unless both Zend HTTP and Laminas HTTP are in use in the same project at which point.

I think we should set all of these state variables (`library_name`, `curl_adapter_typename`, `uri_http_typename`, `http_client`, and `http_client_request`) in both of the `enable` functions rather than relying on them to still have their initial values here.

Reply:
It's not a case of any of the values being set wrongly as everything is being set correctly.  The value is correct, the type is wrong to be passing to [NR_PSTR](https://github.com/newrelic/newrelic-php-agent/blob/main/axiom/nr_axiom.h#L115) which expects a const string not a pointer to a string.  These are the only call to `NR_PSTR` so should be the only place that needs a modification.

Reply:
Ah. TIL. Thanks!